PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/view/tree/ViewNodeTest.java

```diff
@@ -0,0 +1,266 @@
+   */
+  @Test
+  public void testDrawNodeTreeStructureSecondLevel() {
+    ArrayList<ViewNode> rootChildren
+        = (ArrayList) AccessPrivate.getFieldValue("children", ViewNode.class, viewNode);
+    ArrayList<ViewNode> rightChildren
+        = (ArrayList) AccessPrivate.getFieldValue("children", ViewNode.class, rootChildren.get(0));
+    ArrayList<ViewNode> leftChildren
+        = (ArrayList) AccessPrivate.getFieldValue("children", ViewNode.class, rootChildren.get(1));
+    assertEquals(0, rightChildren.size());
+    assertEquals(0, leftChildren.size());
+  }
+
+  @Test
+  public void clickNode() {
+    MouseEvent testMouseEvent = new MouseEvent(viewNode, viewNode, MouseEvent.MOUSE_CLICKED, 0, 0,
+        0, 0, MouseButton.PRIMARY, 0, false, false, false, false, true, false, false, false, false,
+        true, null);
+    viewNode.fireEvent(testMouseEvent);
+    verify(selectionManager, times(1)).select(viewNode);
+  }
+
+  /**
+   * Test of zoomIn method, of class ViewNode. This is mostly based on animation, so we will only
+   * verify that the correct amount of circles is animated.See the testing document in the
+   * documentation folder for an explanation about this and other special cases.
+   */
+  @Test
+  public void testZoomIn() {
+    ArrayList<ViewNode> rootChildren
+        = (ArrayList) AccessPrivate.getFieldValue("children", ViewNode.class, viewNode);
+    Timeline timeline = new Timeline();
+    viewNode.zoomIn(rootChildren.get(1), timeline);
+    assertEquals(3, timeline.getKeyFrames().size());
+  }
+
+  /**
+   * Test of zoomOut method, of class ViewNode. This is mostly based on animation, so we will only
+   * verify that the correct amount of circles is animated. See the testing document in the
+   * documentation folder for an explanation about this and other special cases.
+   */
+  @Test
+  public void testZoomOut() {
+    Timeline timeline = new Timeline();
+    ViewNode zoomLeafL = ViewNode.drawNode(leafL, area, graphPaneSpy, selectionManager);
+    zoomLeafL.zoomOut(timeline);
+    assertEquals(3, timeline.getKeyFrames().size());
+  }
+
+  /**
+   * Test of getGraphArea method, of class ViewNode.
+   */
+  @Test
+  public void testGetGraphArea() {
+    assertEquals(area, viewNode.getGraphArea());
+  }
+
+  /**
+   * Test of getCurrentLeaves method, of class ViewNode.
+   */
+  @Test
+  public void testGetCurrentLeaves() {
+    ArrayList<ViewNode> viewLeaves = viewNode.getCurrentLeaves();
+    assertEquals(2, viewLeaves.size());
+    assertEquals(leafR, viewLeaves.get(0).getDataNode());
+    assertEquals(leafL, viewLeaves.get(1).getDataNode());
+  }
+
+  /**
+   * Test of getClosestParentNode method, of class ViewNode.
+   */
+  @Test
+  public void testGetClosestParentNode() {
+    assertEquals(leafL, viewNode.getClosestParentNode(40, 180).getDataNode());
+  }
+
+  /**
+   * Test of select method, of class ViewNode.
+   */
+  @Test
+  public void testSelect() {
+    // NOT IMPLEMENTED YET
```

Comment:
In the code this method is not implemented yet. The method does however exist, as it is inherited from an interface. This method will make some visual changes to the UI when a node is selected in the future (ex. make the selected node bigger or a different color). I kept the test method because I was too lazy too delete it and later add it when the method is implemented.